src/main/java/com/blog/stream/pagination/PageSpliterator.java

```diff
@@ -112,17 +193,35 @@ public boolean tryAdvance(final Consumer<? super T> action) {
 
         @Override
         public long estimateSize() {
-            return pageSize;
+            return getPageSize();
         }
 
         @Override
         public int characteristics() {
             return ORDERED | IMMUTABLE | SIZED;
         }
 
-        @VisibleForTesting
-        int getPageNumber() {
+        /**
+         * continue advancing until no more pages
+         *
+         * @param action action to invoke on returned items
+         * @return true if can advance
+         */
+        @Override
+        public boolean tryAdvance(final Consumer<? super T> action) {
+            P page = pageFetcher.apply(getPageNumber(), getPageSize());
+            List<T> pageOfItems = itemExtractor.apply(page);
+            pageOfItems.forEach(action);
+            return false;
+        }
+
+        public int getPageSize() {
+            return pageSize;
+        }
+
+        public int getPageNumber() {
```

Comment:
With these getters, my initial thought behind the spliterator objects was that they would only implement the public methods of the Spilterator interface as pageNumber and pageSize are internal details and dont need to be used by other classes.

Reply:
agree, as previous comment, had that in an abstract super class but got rid of it, and never removed.